app/services/companies_details_updator.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class CompaniesDetailsUpdator < ApplicationService
+  def call
+    companies = GPW::Client.new.fetch_gpw_companies_details
+
+    companies.each do |company|
+      Company.find_by(isin: company.fetch(:isin))&.update!(
+        stock: company.fetch(:stock),
+        pbv: company.fetch(:pbv),
+        pe: company.fetch(:pe),
+        condition: company.fetch(:condition),
+      )
+    end
+
+    companies = GPW::Client.new.fetch_nc_companies_details
```

Comment:
I think it would be better to use a single instance of the client:

```ruby
def initialize
  @gpw_client = GPW::Client.new
end
```

and then:

```suggestion
    companies = @gpw_client.fetch_nc_companies_details
```

and the same in line 5.